primitives/inherents/src/lib.rs

```diff
@@ -410,13 +411,24 @@ pub trait ProvideInherent {
 	/// Create an inherent out of the given `InherentData`.
 	fn create_inherent(data: &InherentData) -> Option<Self::Call>;
 
-	/// If `Some`, indicates that an inherent is required. Check will return the inner error if no
-	/// inherent is found. If `Err`, indicates that the check failed and further operations should
-	/// be aborted.
+	/// Determines whether this inherent is required in this block.
+	///
+	/// Ok(None) indicates that this inherent is not required in this block. The default
+	/// implementation returns this.
+	///
+	/// Ok(Some(e)) indicates that this inherent is required in this block. The
+	/// `impl_outer_inherent!`, will call this function from its `check_extrinsics`.
+	/// If the inherent is not present, it will return `e`.
+	///
+	/// Err(_) indicates that this function failed and further operations should be aborted.
 	fn is_inherent_required(_: &InherentData) -> Result<Option<Self::Error>, Self::Error> { Ok(None) }
```

Comment:
at some point we could also link to a related issue: https://github.com/paritytech/substrate/issues/6243

Basically if your pallet have both unsigned transaction and inherents then the `impl_outer_inherent` will just check that one of them is in the block.